src/scheduler/agent/agent.c

```diff
@@ -594,7 +594,34 @@ static void shell_parse(char* confdir, int user_id, int group_id, char* input, c
   (*argv)[idx++] = g_strdup_printf("--groupID=%d", group_id);
   (*argv)[idx++] = "--scheduler_start";
   if (jq_cmd_args)
-    (*argv)[idx++] = jq_cmd_args;
+  {
+    const char *start = jq_cmd_args;
+    const char *current = jq_cmd_args;
+    gboolean in_quotes = FALSE;
+
+    while (*current != '\0')
+    {
+      if (*current == '\'' || *current == '\"')
```

Comment:
Redundant escaping of quotes.

```suggestion
      if (*current == '\'' || *current == '"')
```

Reply:
Done